rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -502,4 +569,108 @@ impl FirmwareProcessor {
 
         (nb, nf)
     }
+
+    /// Read measurement from mailbox and extends it into PCR31
+    ///
+    /// # Arguments
+    /// * `pcr_bank` - PCR Bank
+    /// * `sha384` - SHA384
+    /// * `measurement_log` - Measurement Log
+    /// * `txn` - Mailbox Receive Transaction
+    ///
+    /// # Returns
+    /// * `()` - Ok
+    ///     Err - StashMeasurementReadFailure
+    fn stash_measurement(
+        pcr_bank: &mut PcrBank,
+        sha384: &mut Sha384,
+        measurement_log: &mut StashMeasurementArray,
+        txn: &mut MailboxRecvTxn,
+        log_index: u32,
+    ) -> CaliptraResult<()> {
+        let mut slice: [u8; core::mem::size_of::<StashMeasurement>()] =
```

Comment:
This is more complex than necessary. I suggest:

```rs
  let mut measurement = StashMeasurement::default();
  txn.copy_request(&mut measurement.as_bytes_mut())?;
```

It may also be worth checking that they sent you the correct number of bytes, which you're not currently doing:

```rs
        let mut measurement = StashMeasurement::default();
        if txn.dlen() as usize != measurement.as_bytes().len() {
            return Err(CaliptraError::FW_PROC_STASH_MEASUREMENT_READ_FAILURE);
        }
        txn.copy_request(&mut measurement.as_bytes_mut())?;
```

Reply:
Updated.